core_java/twitter/src/main/java/ca/jrvs/apps/twitter/service/TwitterService.java

```diff
@@ -0,0 +1,164 @@
+  }
+
+  private boolean validateCoordinates(float latitude, float longitude) {
+    // Geolocation check. Lat is +- 90, Long is +- 180
+    return !(Math.abs(latitude) > 90) && !(Math.abs(longitude) > 180);
+  }
+
+  /**
+   * Search a tweet by ID.
+   *
+   * @param id     tweet id
+   * @param fields set fields not in the list to null
+   * @return Tweet object which is returned by the Twitter API
+   * @throws IllegalArgumentException if id or fields param is invalid
+   */
+  @Override
+  public Tweet showTweet(String id, String[] fields) {
+    long longId;
+    try {
+      longId = Long.parseLong(id);
+    } catch (NumberFormatException nfex) {
+      throw new IllegalArgumentException(id + " is not a valid Tweet ID");
+    }
+    Tweet returnedTweet = dao.findById(longId);
+    if (fields.length > 0 && fields[0] != null) {
+      Tweet filteredTweet = new Tweet();
+      Class<Tweet> metaTweet = Tweet.class;
+      Method[] tweetMethods = metaTweet.getMethods();
+      // Obtain the methods in the Tweet class, then check if each is a getter or setter for the
+      // fields passed in. since the fields are private we can't access them directly via reflection
+      for (String field : fields) {
+        Method getter = null;
+        Method setter = null;
+        field = field.substring(0, 1).toUpperCase() + field.substring(1);
+        for (Method m : tweetMethods) {
+          if (m.getName().equals("get" + field)) {
+            getter = m;
+          }
+          if (m.getName().equals("set" + field)) {
+            setter = m;
+          }
+          if (getter != null && setter != null) {
+            try {
+              setter.invoke(filteredTweet, getter.invoke(returnedTweet));
+              break;
+            } catch (IllegalAccessException | InvocationTargetException ex) {
+              throw new IllegalArgumentException("Getter/Setter could not be accessed");
+            }
+          }
+        }
+        if (getter == null || setter == null) {
+          throw new IllegalArgumentException("The field " + field + " does not exist");
+        }
+      }
+      return filteredTweet;
+    }
+    return returnedTweet;
+  }
+
+  /**
+   * Delete Tweet(s) by id(s).
+   *
+   * @param ids tweet IDs which will be deleted
+   * @return A list of Tweets
+   * @throws IllegalArgumentException if one of the IDs is invalid.
+   */
+  @Override
+  public List<Tweet> deleteTweets(String[] ids) {
+    List<Tweet> deletedTweets = new ArrayList<>();
+    for (String id : ids) {
+      long idNum;
+      try {
+        idNum = Long.parseLong(id);
+      } catch (NumberFormatException nfex) {
+        throw new IllegalArgumentException(id + " is not a valid Tweet ID.");
+      }
+      try {
+        deletedTweets.add(dao.deleteById(idNum));
+      } catch (IllegalArgumentException ignored) {
+        // TwitterDao.deleteByID() throws IllegalArgumentException if we can't delete a tweet with
```

Comment:
is this necessary?
if not, please remove and clean up.

Reply:
TwitterDAO throws IllegalArgumentException if a delete fails, but this has no bearing on whether we should continue deleting or not. As per Google Style Guide, if you catch and ignore an exception it must be named "ignored" and you must have a comment explaining why you ignore it. I may have been too wordy there though.

Reply:
please mark it resolved.